pf/tests/internal/providerbuilder/build_provider.go

```diff
@@ -56,3 +62,28 @@ func (impl *Provider) Resources(ctx context.Context) []func() resource.Resource
 	}
 	return r
 }
+
+func EnsureProviderValid(prov *Provider) {
```

Comment:
The name and the signature here are a bit confusing. It is not running any validators from TF (what one might think based on the name). Instead it's filling out some reasonable defaults. 

Question: do we not always want to fill out reasonable defaults for `*Provider`? If so could we consider making this part of a constructor, and refactor struct `p := Provider{}` construction into `p := NewProvider()`?